spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestSparkHoursFunction.java

```diff
@@ -61,13 +68,28 @@ public void testTimestampsNtz() {
     assertThat(scalarSql("SELECT system.hours(CAST(null AS TIMESTAMP_NTZ))")).isNull();
   }
 
+  @TestTemplate
+  public void testBuiltInHourFunction() {
+    String tz = TimeZone.getDefault().getID();
+    assertThat(scalarSql(String.format("SELECT hour(TIMESTAMP '2017-12-01 10:12:55 %s')", tz)))
+        .as("Expected to produce 10")
+        .isEqualTo(10);
```

Comment:
Note: This is how Spark SQL's TIMESTAMP_LTZ support works. The instance in time is converted to the local timezone and the hour function is applied to that. Thus if the instance in time is already specified using the local timezone, when converted the hour hasn't changed, so we know what it is.